packages/backend/src/core/DiscoveryWatcher.ts

```diff
@@ -37,21 +42,24 @@ export class DiscoveryWatcher {
 
   start() {
     this.logger.info('Started')
-    return this.clock.onNewHour(() => {
-      this.taskQueue.addToFront()
+    return this.clock.onNewHour((timestamp) => {
+      this.taskQueue.addToFront(timestamp)
     })
   }
 
-  async update() {
+  async update(timestamp: UnixTime) {
     const metricsDone = initMetrics()
     // TODO: get block number based on clock time
     const blockNumber = await this.provider.getBlockNumber()
-    const timestamp = UnixTime.now()
+    const isDailyReminder = timestamp.equals(
+      timestamp.toStartOf('day').add(9, 'hours'),
```

Comment:
Maybe 8 here? This is Unix Time
```suggestion
      timestamp.toStartOf('day').add(8, 'hours'),
```